Better handling of typeset/declare

algorythmic opened the pull request:
* complete variable names when appropriate
* complete more than one var/fun name
* complete `+` options in addition to `-` options
* recognize `-F` in addition to `-f`

scop commented:
Could you add test cases for your improvements to the test suite? See existing stuff in `test/` and CONTRIBUTING.md. I've just pushed 81f2708198dd87604c4257b94b18c0e96a2852fb; `test/lib/completions/declare.exp` is the file to add test cases for this stuff to.

algorythmic commented:
Added the tests and fixed up the pull request.

The failing tests seem unrelated:

    Running ./completion/kill.exp ...
    FAIL: kill 1 should show completions at timeout
    ...
    Running ./completion/renice.exp ...
    FAIL: Tab should show complete process ids at timeout

whereas

    Running ./completion/declare.exp ...
    PASS: declare - should show completions
    PASS: declare + should show completions
    PASS: declare -p BASH_ARG should show completions
    PASS: declare -f _parse_ should show completions

scop commented:
Yes, the test suite is unfortunately a bit unstable, and those do look unrelated. I've restarted the run and will check back later.
